linera-execution/src/wasm/system_api.rs

```diff
@@ -327,6 +327,16 @@ where
             .map_err(|error| RuntimeError::Custom(error.into()))
     }
 
+    /// Panics if the current time at block validation is `>= timestamp`. Note that block
+    /// validation happens at or after the block timestamp, but isn't necessarily the same.
```

Comment:
Nit: this doesn't actually `panic`, right? I think you mean that the contract panics? If that's the case, this should be said in the `ContractRuntime` documentation, and maybe here we could say "Aborts execution if..." :thinking: 

```suggestion
    /// Rejects the transaction if the current time at block validation is `>= timestamp`. Note
    /// that block validation happens at or after the block timestamp, but isn't necessarily the
    /// same.
```